make_prg/from_msa/cluster_sequences.py

```diff
@@ -0,0 +1,199 @@
+import logging
+from collections import defaultdict
+from typing import List, Dict, Iterator, Union
+from itertools import starmap, repeat, chain
+from collections import Counter
+import time
+
+import numpy as np
+from sklearn.cluster import KMeans
+
+from make_prg.from_msa import MSA
+from make_prg.seq_utils import ungap
+
+Sequence = str
+Sequences = List[str]
+IDs = List[str]
+SeqToIDs = Dict[Sequence, IDs]
+SeqToSeqs = Dict[Sequence, Sequences]
+ClusteredIDs = List[IDs]
+ClusteredSeqs = List[Sequences]
+KmerIDs = Dict[Sequence, int]
+
+DISTANCE_THRESHOLD: float = 0.2
+LENGTH_THRESHOLD: int = 5
+MAX_CLUSTERS: int = 10
+
+
+def count_distinct_kmers(seqs: Sequences, kmer_size: int) -> KmerIDs:
+    for seq in seqs:
+        if len(seq) < kmer_size:
+            raise ValueError(f"Input sequence {seq} has length < kmer size {kmer_size}")
+    result = dict()
+    kmer_id = 0
+    for seq in seqs:
+        for start_pos in range(len(seq) - kmer_size + 1):
+            kmer = seq[start_pos : start_pos + kmer_size]
+            if kmer not in result:
+                result[kmer] = kmer_id
+                kmer_id += 1
```

Comment:
This could all be replaced with a [`Counter`](https://docs.python.org/3/library/collections.html#collections.Counter)?

Reply:
Not trivially i don't think? So that particular code assigns an ID to each found kmer. then in `count_kmer_occurrences` we make a count matrix, fill it up with zeros for each sequence and each kmer, and increment counts using the IDs produced here.

How would you use `Counter`, would you count kmers in each sequence and then merge them somehow? you'd need a matrix where each column is a given kmer

Reply:
Ah ok. I admit I didn't fully understand what was happening here. I was merely suggesting it looked like there might have been scope for it. But it sounds like there isn't